src/sagemaker/estimator.py

```diff
@@ -158,9 +154,14 @@ def fit(self, inputs, wait=True, logs=True, job_name=None):
             base_name = self.base_job_name or base_name_from_image(self.train_image())
             self._current_job_name = name_from_base(base_name)
 
-        # if output_path was specified we use it otherwise initialize here
+        # if output_path was specified we use it otherwise initialize here.
+        # For Local Mode with no_internet=True we don't need an explicit output_path
         if self.output_path is None:
-            self.output_path = 's3://{}/'.format(self.sagemaker_session.default_bucket())
+            no_internet = get_config_value('local.no_internet', self.sagemaker_session.config)
+            if self.sagemaker_session.local_mode and no_internet:
```

Comment:
If we start throwing more configuration into local_mode, then we should have a local_mode configuration object in place of the local_mode boolean variable. I think this is okay now, but once we get one more piece of config, let's refactor this.

Reply:
sounds good.